Engine/source/terrain/hlsl/terrFeatureHLSL.cpp

```diff
@@ -918,13 +953,12 @@ ShaderFeature::Resources TerrainMacroMapFeatHLSL::getResources( const MaterialFe
       // If this is the first detail pass then we 
       // samples from the layer tex.
       res.numTex += 1;
-   }
+      res.numTexReg += 1;
```

Comment:
This could use a comment in what we are accounting for

Reply:
This is basically just me duplicating what the detail feature currently does -- I'm not 100% sure why it was written this way (+1-ing those vars twice), but my understanding is that with deferred rendering we no longer need to count every instance of a detail or macro texture against the registers limit. For detail it just adds 2 to the limit if the terrain has any detail layers at all, so I did the same thing for macro.

If you count every detail or macro map against the reg list, you can only have like 4 of either in a cell before the shader builder will start stripping layers out. It has a limit of 10 for texreg

Reply:
We switched to a single pass approach so we aren’t limited by texture count, since we pack it into an array.

but you need to count that we add the macromap array, so I’m guessing this is too count that